### DISCUSSION TITLE
MyWhoosh setup, please help

### DISCUSSION THREAD

**Discussion by Vendo233** [cagnulein/qdomyos-zwift#2521] [Q&A]
Hello

I have following devices 

- Echelon GT
- Garmin fenix 6x Pro
- Garmin ANT+ old HR chest strap
- Android phone: Galaxy Edge S7, Pixel 4, Pixel 7 ( also have some old phone like iPhone 6s or iPhone X but did not buy app for iOS yet ) 
- Android table: Galaxy Tab A9+
- Raspbery pi 2 W ( not used, just listing if this would be better bridge than Android phone ) 

I would like to connecy Echelon GT and my bike ride to MyWhoosh including heart rate information if possible.

What works:
- Combination of Pixel 7 and Galaxy Tab A9+, I can get cadance into my WHoosh and I can complete the ride.
- I can connect fenix 6x Pro ompanion HR app to Pixel 7 ( my main phone ) and get HR into QZ.
- I can connect echelon GT into Pixel 7 including HR from fenix 6x pro using companion app. 
- I can connect echelon GT into Galaxy S7 including HR from ANT+ strap. ( I thing S7 supports ANT+ directly ) 

What does not work.
- Echelon can not connect to pixel 4 at all , nothing, it is like the BT chipset is not supported.
- I can not pass HR from QZ to MyWHoosh using Pixel 7 and A9+ tablet . 
- I can not pass cadance or anything from Galaxy S7 to MyWhoosh on the tablet. I can see and pair the phone in the app but the cadance is not connected and Wattage stays zero, can not ride. 

I did play with the expeprimental options in the Android APP menu but did not see any improvement. 

I want to try the Raspebry Pi method next. Any advise would be welcome.

**Comment by cagnulein** [comment]
> Echelon can not connect to pixel 4 at all , nothing, it is like the BT chipset is not supported.

do you have both bluetooth and location services enabled on the pixel 4?

> I can not pass HR from QZ to MyWHoosh using Pixel 7 and A9+ tablet .

this is a known limitation of mywhoosh: they don't handle power and heart rate from the same device. ask them to add this!

> I can not pass cadance or anything from Galaxy S7 to MyWhoosh on the tablet. I can see and pair the phone in the app but the cadance is not connected and Wattage stays zero, can not ride.

rename the s7 device to bike (only 4 letters) and reboot it

let me know

**Comment by Vendo233** [comment]
Hello Roberto, here is my feedback.

QZ app: 2.16.66

Pixel 4
- location and BT are both enables and it is def. nothing. I can try downgrading it to Android 12 or 11. curently in 13.  Will see if that makes any difference.

Galaxy Edge S7, Motorola E6 or NExus 5x
- changed name to "bike" on all
- all these old guys can not pass Cadance. The motorola E6 first time  paired all sensors ( HR, Cadance , Control and W ) but then second time and later just Watage and HR. See pictures. 
![image](https://github.com/user-attachments/assets/d09735d3-daab-42c7-889f-9511097515a1)

- I found out that when i click on Controlabe icon I can pair bike and it becomes green ( manually ) , but cadance is blank and I can not start the ride. Pairing manually yield nothing. 

Raspbery Pi 2W
- the "make" compilation starts it runs for some time and then the device hangs, I can not SSH , have to hard restart the Pi and run Make again. I think that is not have it should be. tried "make clean" it hangs. Unable to finish the compilarion.
- Would it be possible to share already pre-compiled version for RPi 2 W?

Picel 7 
- works ok still, pairing makes Cadance and Comtrolable and W green ( not HR ) but it is my prime phone and I need to figure out how to setup the Echelon e/o my prime phone for my wife.
- btw ONCE , very first time the HR worked in My Whoosh from my fenix 6 via companion app. it worked just once., never again since. 

I have also iPhone X laying around and Iphone 7, would those devices be better bridges for MyWhoosh?

**Comment by Vendo233** [comment]
Pixel 4 update

- flashed A12 , installed your latest app 2.16.68.
- installed garmin and pair my Fenix 6x

And mywhoos is fully operational 

![PXL_20240820_034130977 MP](https://github.com/user-attachments/assets/361b76a8-1ff9-473a-b914-41f06e222514)
![PXL_20240820_034049200](https://github.com/user-attachments/assets/2c5c196a-74ee-4be4-a722-d7686b9ed61d)

**Comment by Vendo233** [comment]
![PXL_20240820_034044056](https://github.com/user-attachments/assets/04de1770-c10b-4e7e-b031-ea55d12adf77)

**Comment by cagnulein** [comment]
> Hello Roberto, here is my feedback.
> 
> 
> 
> QZ app: 2.16.66
> 
> 
> 
> Pixel 4
> 
> - location and BT are both enables and it is def. nothing. I can try downgrading it to Android 12 or 11. curently in 13.  Will see if that makes any difference.
> 
> 

so i guess we are ok with pixel 4 right? what did you do to make it work?

> 
> Galaxy Edge S7, Motorola E6 or NExus 5x
> 
> - changed name to "bike" on all
> 
> - all these old guys can not pass Cadance. The motorola E6 first time  paired all sensors ( HR, Cadance , Control and W ) but then second time and later just Watage and HR. See pictures. 
> 
> ![image](https://github.com/user-attachments/assets/d09735d3-daab-42c7-889f-9511097515a1)
> 
> 
> 
> - I found out that when i click on Controlabe icon I can pair bike and it becomes green ( manually ) , but cadance is blank and I can not start the ride. Pairing manually yield nothing. 
> 
> 

it could be a problem on the bluetooth chip of the device connecting and streaming data on the same time. don't you see a wahoo virtual device on mywhoosh as well?

> 
> Raspbery Pi 2W
> 
> - the "make" compilation starts it runs for some time and then the device hangs, I can not SSH , have to hard restart the Pi and run Make again. I think that is not have it should be. tried "make clean" it hangs. Unable to finish the compilarion.

you went out of ram. increase the swap 

> 
> - Would it be possible to share already pre-compiled version for RPi 2 W?
> 
> 

you can ask it here https://github.com/cagnulein/qdomyos-zwift/issues/2499#issuecomment-2297441103
he just did it!

> 
> Picel 7 
> 
> - works ok still, pairing makes Cadance and Comtrolable and W green ( not HR ) but it is my prime phone and I need to figure out how to setup the Echelon e/o my prime phone for my wife.
> 
> - btw ONCE , very first time the HR worked in My Whoosh from my fenix 6 via companion app. it worked just once., never again since. 
> 
> 

mywhoosh needs to work on this, this is for sure a problem on their end

> 
> I have also iPhone X laying around and Iphone 7, would those devices be better bridges for MyWhoosh? 

yes for sure, iphone are always better on bluetooth 

let me know!

**Reply by Vendo233** [reply]
> it could be a problem on the bluetooth chip of the device connecting and streaming data on the same time. don't you see a wahoo virtual device on mywhoosh as well?

on Pixel 7 I can see 3 sensors 1x Pixel 7 , 2x Wahoo
on Pixel 4 and older phones only phone sensor no wahoo ( turned ON in QZ )

**Comment by cagnulein** [comment]
Ok so use the pixel 7 with the wahoo device. Enable the mywhoosh setting in
the experimental settings as well

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mar 20 ago 2024 alle 19:13 Vendo233 ***@***.***> ha
scritto:

**Comment by Vendo233** [comment]
Roberto

I did increse the SWAP on RPI2W and it completed the compilation in about 45min.
After that I was able to commect Echelon to the RPI and RPI to MyWhoosh.

![image](https://github.com/user-attachments/assets/023c0685-f0a0-470c-8a65-ce4ebc907c29)

now I couldn`t find how steps to connect via VNC or any other way to access settings? 
I would like to enable Garmin Fenix 6x for HR on RPI or my old ANT+ chest strap if it is possible. 
apology for so many question.

**Reply by cagnulein** [reply]
Great! For vnc did you check this https://www.pitunnel.com/doc/access-vnc-remote-desktop-raspberry-pi-over-internet ?

You can connect the fenix 6 only if it has Bluetooth heart rate broadcasting mode. Check it in the settings

**Reply by Vendo233** [reply]
> Great! For vnc did you check this https://www.pitunnel.com/doc/access-vnc-remote-desktop-raspberry-pi-over-internet ?
> 
> You can connect the fenix 6 only if it has Bluetooth heart rate broadcasting mode. Check it in the settings

Thank you Roberto, If I installed light version of RaspOS can I still use VNC? I`m getting " Cannot currently show desktop"

**Comment by cagnulein** [comment]
apt install vncserver

or similar should add it

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mer 21 ago 2024 alle 21:12 Vendo233 ***@***.***> ha
scritto:

**Reply by Vendo233** [reply]
> apt install vncserver or similar should add it Roberto Viola Software engineer and open source enthusiast http://robertoviola.cloud Il giorno mer 21 ago 2024 alle 21:12 Vendo233 ***@***.***> ha scritto:
> […](#)
> Great! For vnc did you check this https://www.pitunnel.com/doc/access-vnc-remote-desktop-raspberry-pi-over-internet ? You can connect the fenix 6 only if it has Bluetooth heart rate broadcasting mode. Check it in the settings Thank you Roberto, If I installed light version of RaspOS can I still use VNC? I`m getting " Cannot currently show desktop" — Reply to this email directly, view it on GitHub <[#2521 (reply in thread)](https://github.com/cagnulein/qdomyos-zwift/discussions/2521#discussioncomment-10411806)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AAALYWBFRWTT6IQIQGRW663ZSTRBBAVCNFSM6AAAAABMVZOK5SVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTANBRGE4DANQ> . You are receiving this because you commented.Message ID: ***@***.*** com>

it seems I will need to install desktop environment on the Raspberry to get the GUI, start QZ with GUI and there change settings?
Fenix 6X Pro can broadcast HR, but Raspberry is not picking it up. Hoping there is something in GUI setting which needs to be turned ON to enable that?

**Comment by cagnulein** [comment]
there is a way to have vnc via web, i will search the post.
yes you have to configure hr to the UI

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mer 21 ago 2024 alle 21:27 Vendo233 ***@***.***> ha
scritto:

**Comment by cagnulein** [comment]
@Vendo233  this https://github.com/cagnulein/qdomyos-zwift/discussions/1685#discussioncomment-7382978

**Comment by Vendo233** [comment]
Roberto, when do you think we can see Android version 2.16.68 in App store? 
the 2.16.66 does not broadcast HR to MyWHoosh, while 2.16.68 does.

**Comment by cagnulein** [comment]
Perfect, I'm pushing to the beta channel this weekend and then Google will
take another week to review it. You can use the apk from here in the
meantime

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno gio 22 ago 2024 alle 17:44 Vendo233 ***@***.***> ha
scritto:

**Comment by Vendo233** [comment]
Roberto, I`m using Pixel 4 with QZ app and MyWhoosh, all works great! but... 

I`m finding my self pedaling like crazy in my workout, basicaly spinning in the MyWhoosh due to high cadance ( I think ) my wattage is under 1w/kg almost all the time. I`m probably weak cyclist but is there a way to tweek the QZ to go little faster in MyWhoosh? I`m doing just free rides, no competition. Just to make it easier little bit ( 10% )

**Reply by cagnulein** [reply]
@Vendo233 Which bike do you have?

**Reply by Vendo233** [reply]
> @Vendo233 Which bike do you have?

Yes, forgot to mention, Echolon GT

**Comment by cagnulein** [comment]
Wheel on the back right? Zwift resistance offset setting is set to 18 on qz?

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mer 16 ott 2024 alle 21:37 Vendo233 ***@***.***> ha
scritto:

**Reply by Vendo233** [reply]
this guy

![image](https://github.com/user-attachments/assets/a567e933-4cba-4461-854e-9e98daed5fa8)

**Reply by cagnulein** [reply]
Hah so it doesn't have auto resistance so you have to increase the resistance manually. You can follow the target resistance tile to know which one set

Let me know

**Reply by Vendo233** [reply]
> Hah so it doesn't have auto resistance so you have to increase the resistance manually. You can follow the target resistance tile to know which one set
> 
> Let me know

Correct, and it works perfect, if I need to climb I have to cranck up the resistance.
Just was wondering if there is a setting in QZ which could make it little easier

**Comment by cagnulein** [comment]
No it's not about qz, it's about mywhoosh. They require X wattage to climb,
you can increase the wattage calibration but it's like cheating so I don't
suggest this
Do you see the wattage increase when you are increasing the resistance?

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mer 16 ott 2024 alle 22:02 Vendo233 ***@***.***> ha
scritto:

**Comment by Vendo233** [comment]
Yes ,I do see it. It works as it should, just it feels hard and feels wrong
when you see others climbing with 3-4 W/kg to the steap hill and passing me
by :-)

No problem, soo many options in the settings in the qz so just wanted to
ask.

**Comment by cagnulein** [comment]
Eheheh 3w/kg it's a lot, so don't feel this, everyone has its training
status. Take it easy and enjoy your workouts!

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mer 16 ott 2024 alle 22:10 Vendo233 ***@***.***> ha
scritto:
